File: DuDoong-Api/src/main/java/band/gosrock/api/example/dto/ExampleResponse.java
package band.gosrock.api.example.dto;


import band.gosrock.domain.domains.example.domain.ExampleEntity;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class ExampleResponse {

    private final Long id;
    private final String content;

    public static ExampleResponse from(ExampleEntity exampleEntity) {
        return new ExampleResponse(exampleEntity.getId(), exampleEntity.getContent());
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/example/dto/ExampleResponse.kt
package band.gosrock.api.example.dto

import band.gosrock.domain.domains.example.domain.ExampleEntity

data class ExampleResponse(
    val id: Long,
    val content: String,
) {
    companion object {
        @JvmStatic
        fun from(exampleEntity: ExampleEntity): ExampleResponse =
            ExampleResponse(
                id = exampleEntity.id,
                content = exampleEntity.content,
            )
    }
}
